Documentation/coding-guidelines/adding-api-guidelines.md

```diff
@@ -0,0 +1,124 @@
+#Add APIs
+
+- [Determining versions and targets](#determining-versions-and-targets)
+- [Making the changes in repo](#making-the-changes-in-repo)
+- [FAQ](#faq)
+
+##Determining versions and targets
+
+1. [Determine what library](#determine-what-library) the API goes into.
+2. [Determine the target framework](#determine-target-framework) for the library that will contain the API.
+3. [Determine the version](#determine-library-version) for the library that will contain the API.
+
+###Determine what library
+- Propose a library for exposing it as part of the [API review process](http://aka.ms/apireview).
+- Keep in mind the API might be exposed in a reference assembly that
+doesn't match the identity of the implementation. There are many reasons for this but
+the primary reason is to abstract the runtime assembly identities across
+different platforms while sharing a common API surface and allowing us to refactor
+the implementation without compat concerns in future releases.
+
+###Determine target framework
+`<latest>` is the version currently under development.
+
+- If the library is [part of netstandard](#isnetstandard)
```

Comment:
For before netstandard2.0 you are part of netstandard if you are adding the API to an assembly that is in the set of assemblies included in netstandard. For netstandard2.0 itself it is only if the type is netstandard.dll.